lntest/harness.go

```diff
@@ -670,8 +670,10 @@ func (n *NetworkHarness) EnsureConnected(ctx context.Context,
 //
 // NOTE: This function may block for up to 15-seconds as it will not return
 // until the new connection is detected as being known to both nodes.
-func (n *NetworkHarness) ConnectNodes(ctx context.Context, t *testing.T,
-	a, b *HarnessNode) {
+func (n *NetworkHarness) ConnectNodes(t *testing.T, a, b *HarnessNode) {
+	ctxb := context.Background()
+	ctx, cancel := context.WithTimeout(ctxb, DefaultTimeout)
```

Comment:
Just a thought: I think ideally we could have a different timeout context for the first part of this function vs the last part in the `Predicate`. Although the whole block should finish in `DefaultTimeout` normally.

Reply:
I think so too. In the future we may fine tune all these timeout values.